### PR TITLE
Fix several bugs relating to locking out non-backend URLs to login

### PR DESCRIPTION
**Pull request by CorySanin**
fix empty settings bug, fix expression not defined bug, fixed logged out template

### CODE REVIEW

**Comment by CorySanin** [comment]
Can I get a review on this?

File: castle/cms/browser/templates/secure-login.pt [hunk 1]
```diff
@@ -10,7 +10,7 @@
       tal:define="utils context/@@castle-utils;
                   site_login python: hasattr(context,'portal_registry');
                   portreg python: context.portal_registry if site_login else {'plone.backend_url':[],'plone.only_allow_login_to_backend_urls':False};
-                  backend_urls python:portreg['plone.backend_url'];
+                  backend_urls python:portreg['plone.backend_url'] or [];
```
Comment (tkimnguyen) [inline castle/cms/browser/templates/secure-login.pt]: Does this cover all situations, e.g. empty string and None?

File: castle/cms/skins/castle/login_form.cpt [hunk 2]
```diff
@@ -60,7 +58,10 @@
                         target request/target|nothing;
                         target python:test(target in ('_parent', '_top', '_blank', '_self'), target, None);
                         ztu modules/ZTUtils;
-                        login nocall: context/@@secure-login">
+                        login nocall: context/@@secure-login;
+                        backend_urls python:context.portal_registry['plone.backend_url'] or [];
```
Comment (tkimnguyen) [inline castle/cms/skins/castle/login_form.cpt]: a shame there isn't a way to define this expression in just one place and reuse

File: castle/cms/browser/templates/secure-login.pt (hunk 1 above)
Comment (CorySanin) [reply castle/cms/browser/templates/secure-login.pt]: While it shouldn't ever happen, if it's an empty string, leng('') will still return 0. That's fine.
If it's None, it defaults to [].

File: castle/cms/skins/castle/login_form.cpt [hunk 3]
```diff
@@ -60,7 +58,10 @@
...
+                        only_allow_login_to_backend_urls python:context.portal_registry['plone.only_allow_login_to_backend_urls'];
+                        bad_domain python: only_allow_login_to_backend_urls and len(backend_urls) > 0 and portal_url.rstrip('/') not in backend_urls;">
```
Comment (tkimnguyen) [inline castle/cms/skins/castle/login_form.cpt]: I know I wrote this expression originally but it's kinda ugly... is it possible to improve?

**Pull request merged**